### PR TITLE
Declare many of our remotables

### PR DESCRIPTION
**Pull request by erights**
To help track down the mysterious missing `iface` strings.

To make progress on https://github.com/Agoric/agoric-sdk/issues/804

To make progress on https://github.com/Agoric/agoric-sdk/issues/2018

### CODE REVIEW

File: packages/zoe/src/contractFacet/contractFacet.js [hunk 1]
```diff
@@ -354,7 +354,9 @@ export function buildRootObject(powers, _params, testJigSetter = undefined) {
           details`invitations must have a description string: ${description}`,
         );
 
-        const invitationHandle = /** @type {InvitationHandle} */ (harden({}));
+        const invitationHandle = /** @type {InvitationHandle} */ (makeHandle(
+          'InvitationHandle',
+        ));
```
Comment (michaelfig) [inline packages/zoe/src/contractFacet/contractFacet.js]:
```suggestion
        const invitationHandle = makeHandle('Invitation');
```

And possibly need to `@typedef {Handle<'Invitatiton'>} InvitationHandle` if it doesn't work without the typecast.

File: packages/zoe/src/makeHandle.js [hunk 2]
```diff
@@ -11,5 +12,5 @@ export const makeHandle = handleType => {
   // This assert ensures that handleType is referenced.
   assert.typeof(handleType, 'string', 'handleType must be a string');
   // Return the intersection type (really just an empty object).
-  return /** @type {Handle<H>} */ (harden({}));
+  return /** @type {Handle<H>} */ (Remotable(`Alleged: ${handleType}`));
```
Comment (michaelfig) [inline packages/zoe/src/makeHandle.js]:
```suggestion
  return /** @type {Handle<H>} */ (Remotable(`Alleged: ${handleType}Handle`));
```

**Review by michaelfig — APPROVED** [review]
Take my comments with a grain of salt, but eliminating typecasts is my main goal.

**Comment by erights** [comment]
Hi @michaelfig this is now ready for review. Thanks.

**Comment by erights** [comment]
This is now blocked on #2207 which explains the three failures reported by CI.

**Comment by erights** [comment]
Hi @michaelfig I made a lot of changes since you approved. Could you review again? Thanks.

**Review by michaelfig — APPROVED** [review]
A lot of these labels seem not to be strictly necessary, as they are not empty objects.  Is there a reason to add them all?

Otherwise, LGTM.

**Review by FUDCo — COMMENTED** [review]
This is not what I was expecting.  I had, perhaps mistakenly, framed the problem as "empty objects are treated as presences rather than data, which makes them dangerous when they are actually data" and I assumed the fix was to change serialization to always treat empty objects as data unless explicitly declared otherwise.  Making this change requires us to first label all the empty objects that are currently being used as presences (mainly as identity markers, if I understand correctly), which is what I thought you were doing.  This is why I was surprised to hear that there were many cases of things that needed to be so declared in SwingSet, since it doesn't (as best I can recall) make much, if any, use of the empty-object-passed-for-its-identity pattern.  But here I see this PR declaring lots of `Remotable`s that are _not_ empty objects, which, while it might eventually be useful from a documentary perspective, seems unrelated to the underlying problem I thought we were trying to fix.  Instead, it seems like we're just adding a lot of boilerplate noise with little payoff.

Also, I think the word "Alleged" as used here has jumped the shark.  If all `Remotable` types are alleged types (are they?), then labelling each one as such is empty ceremony -- it takes up space but the reader's brain learns to ignore it since it conveys no new information.

**Comment by erights** [comment]
Ok @FUDCo , responded as we discussed. Ready for your rereview. Thanks.

**Review by FUDCo — APPROVED** [review]
That's nicer.

**Pull request merged**